### DISCUSSION TITLE
v2.2.1 - Community Builds Thread

### DISCUSSION THREAD

**Discussion by gianni-rosato** [gianni-rosato/svt-av1-psy#66] [Show and tell]
# Community Builds Thread

This is a place for the community to share unofficial tools not affiliated with the project -- mainly consisting of binaries compiled by community members.

## Trust

:warning: **Community-built binaries are not affiliated with the SVT-AV1-PSY project & are to be used at your own risk.** If you encounter bugs while using a community-built binary, ensure that you double-check with the person responsible for providing you with the binary before submitting an issue report.

## Architecture

When downloading pre-compiled binaries, you might see AVX, AVX2, AVX-512, x86-64-v3, etc. If you don't know exactly what ISA extensions your CPU supports, here is a chart to help you quickly understand your hardware's support:

### AMD (Desktop)
```ocaml
Desktop Gen       Architecture       Build
Ryzen 1000        AVX2               x86-64-v3
Ryzen 2000        AVX2               x86-64-v3
Ryzen 3000        AVX2               x86-64-v3
Ryzen 5000        AVX2               x86-64-v3
Ryzen 7000        AVX-512            x86-64-v4
```
### Intel (Desktop)
```js
Desktop Gen                 Archit.  Build
Haswell (4000-series)       AVX2     x86-64-v3
Broadwell (5000-series)     AVX2     x86-64-v3
Skylake (6000-series)       AVX2     x86-64-v3
Kaby Lake (7000-series)     AVX2     x86-64-v3
Coffee Lake (8/9000-series) AVX2     x86-64-v3
Comet Lake (10000-series)   AVX2     x86-64-v3
Rocket Lake (11000-series)  AVX-512  x86-64-v4
Alder Lake (12000-series)   AVX2     x86-64-v3
Raptor Lake (13/14000-ser.) AVX2     x86-64-v3
```

It is also helpful to know the various options available with C compilers, like what is available when specifying `-march` & `-mtune` on x64 CPUs:

Known valid x64 arguments for `-march=`:
> i386 i486 i586 pentium lakemont pentium-mmx winchip-c6 winchip2 c3 samuel-2 c3-2 nehemiah c7 esther i686 pentiumpro pentium2 pentium3 pentium3m pentium-m pentium4 pentium4m prescott nocona core2 nehalem corei7 westmere sandybridge corei7-avx ivybridge core-avx-i haswell core-avx2 broadwell skylake skylake-avx512 cannonlake icelake-client rocketlake icelake-server cascadelake tigerlake cooperlake sapphirerapids emeraldrapids alderlake raptorlake meteorlake graniterapids graniterapids-d arrowlake arrowlake-s lunarlake pantherlake bonnell atom silvermont slm goldmont goldmont-plus tremont gracemont sierraforest grandridge clearwaterforest knl knm intel geode k6 k6-2 k6-3 athlon athlon-tbird athlon-4 athlon-xp athlon-mp x86-64 x86-64-v2 x86-64-v3 x86-64-v4 eden-x2 nano nano-1000 nano-2000 nano-3000 nano-x2 eden-x4 nano-x4 lujiazui yongfeng k8 k8-sse3 opteron opteron-sse3 athlon64 athlon64-sse3 athlon-fx amdfam10 barcelona bdver1 bdver2 bdver3 bdver4 znver1 znver2 znver3 znver4 znver5 btver1 btver2 generic native

Known valid x64 arguments for `-mtune=`:
> generic i386 i486 pentium lakemont pentiumpro pentium4 nocona core2 nehalem sandybridge haswell bonnell silvermont goldmont goldmont-plus tremont sierraforest grandridge clearwaterforest knl knm skylake skylake-avx512 cannonlake icelake-client icelake-server cascadelake tigerlake cooperlake sapphirerapids alderlake rocketlake graniterapids graniterapids-d arrowlake arrowlake-s pantherlake intel lujiazui yongfeng geode k6 athlon k8 amdfam10 bdver1 bdver2 bdver3 bdver4 btver1 btver2 znver1 znver2 znver3 znver4 znver5

I'll leave it to you all to look up which options work for your hardware.

- Note: `-march=foo` implies `-mtune=foo` unless you also specify a different `-mtune`. This is one reason why using `-march` is better than just enabling options like `-mavx` without doing anything about tuning.

## Antivirus

Be wary of antivirus software on Windows detecting EXEs distributed here as malicious software. While they may not always legitimately be malicious, it is important to maintain a healthy level of skepticism when running code that someone else has compiled.

**Comment by Uranite** [comment]
svt-av1-psy v2.2.1
x86-64-v3
```
cmake --fresh -B svt_build -T ClangCL -DENABLE_AVX512=OFF -DBUILD_SHARED_LIBS=OFF -DCMAKE_CXX_FLAGS_RELEASE="-flto /DNDEBUG /O2 -march=x86-64-v3" -DCMAKE_C_FLAGS_RELEASE="-flto /DNDEBUG /O2 -march=x86-64-v3"
```
[SvtAv1EncApp.zip](https://github.com/user-attachments/files/16827182/SvtAv1EncApp.zip)

**Comment by PorcelainSky** [comment]
There seems to be a problem with the released Windows build,After running for a few seconds, an error occurs.

**Reply by Uranite** [reply]
Are you talking about my build or the one in github releases?

**Reply by shuniichan** [reply]
> Are you talking about my build or the one in github releases?

Your build works fine while the one in official github release gives error messages for me.

**Reply by Uranite** [reply]
It might be that your cpu doesn't support AVX512, after all, only a handful of cpu is x86-64-v4, such as Ryzen 7000, Ryzen 9000, and Intel 11th gen

**Reply by PorcelainSky** [reply]
> It might be that your cpu doesn't support AVX512, after all, only a handful of cpu is x86-64-v4, such as Ryzen 7000, Ryzen 9000, and Intel 11th gen

It is official. As you said, my CPU does not support avx512, so it causes errors.

**Comment by mmkzer0** [comment]
svt-av1-psy (MacOS 15 Static) v2.2.1 7077c9f
Compiled by Reaven aka mmkzero
- targets M3 arch
- compiled with LLVM clang 18.1.8
- LTO and PGO Optimized

build arguments used:
`./build.sh --native --static --release --enable-lto --asm=yasm cc=/opt/homebrew/opt/llvm/bin/clang cxx=/opt/homebrew/opt/llvm/bin/clang++ --verbose --enable-pgo pgo-dir=~/dev/shm pgo-compile-use`

SHA-512 Checksum:
`ac54eb03b1ff3eef048f8928d1ec60967c36fe64dc56d86a8deca4dff495c696e62c6d1ae85c047d6e03223eda95780c1eacb50db5e9067adc224343cc32718c  SvtAv1EncApp`

`50d273339fe9d41ef6fa2dbe802ee903c88984565cb8b70a48a55c8cc41003af855eafd6e29827c372a521b69a67c81afb30ae4fa03ef3a2de26bb440c4821e5  SvtAv1EncApp-M3.zip`

[SvtAv1EncApp-M3.zip](https://github.com/user-attachments/files/16917685/SvtAv1EncApp-M3.zip)

**Comment by NextWork123** [comment]
svt-av1-psy (Windows) v2.2.1 7077c9ff6f638328f6a5139fafcc2d8d537ebb77
Compiled by NextWorks

- Compiled with LLVM clang 19.1.0-rc4
- Compiled with -03 and LTO and x86-64-v4 with AVX512 enabled

`cmake --fresh -B svt_build -G Ninja -DCMAKE_C_COMPILER="C:/Users/Administrator/.utils/llvm-mingw/bin/clang-19.exe" -DCMAKE_CXX_COMPILER="C:/Users/Administrator/.utils/llvm-mingw/bin/clang++.exe" -DENABLE_AVX512=ON -DBUILD_SHARED_LIBS=OFF -DCMAKE_CXX_FLAG
[SvtAv1EncApp.zip](https://github.com/user-attachments/files/16923479/SvtAv1EncApp.zip)
S_RELEASE="-flto -DNDEBUG -O3 -march=x86-64-v4" -DCMAKE_C_FLAGS_RELEASE="-flto -DNDEBUG -O3 -march=x86-64-v4" `

SHA-512 Checksum: 
`7d593781e99bc07804c95b2b2be5c5260ac75eff4293cd663b960265fa592eb3cd4f67d99631e40d791a4dde76d3be614a32cc8a1455c40e722f8b8263932568  SvtAv1EncApp`

SHA-256 Checksum:
`97d43fb160bb45c52bed6570d8b6da38f60e800434c2d1b34d73d006c083b2ef SvtAv1EncApp`

SHA-1 Checksum:
`665e9c1136a8c542428bdc33aba5b4b94574c198 SvtAv1EncApp`

MD5 Checksum:
`a123f3f460536e84f601998415c8488c SvtAv1EncApp`

[SvtAv1EncApp.zip](https://github.com/user-attachments/files/16923481/SvtAv1EncApp.zip)

**Comment by Uranite** [comment]
svt-av1-psy v2.2.1 aarch64 for Termux
compiled via docker in the termux-packages repo with modified svt-av1 build.sh
extract it and install the .deb file via `dpkg -i`
[svt-av1_2.2.1_aarch64.zip](https://github.com/user-attachments/files/17001063/svt-av1_2.2.1_aarch64.zip)

**Comment by Uranite** [comment]
svt-av1-psy v2.2.1 for windows
generic build (no arch set)
`cmake --fresh -B svt_build -T ClangCL -DBUILD_SHARED_LIBS=OFF -DCMAKE_CXX_FLAGS_RELEASE="-flto /DNDEBUG /O2" -DCMAKE_C_FLAGS_RELEASE="-flto /DNDEBUG /O2"`
[SvtAv1EncApp.zip](https://github.com/user-attachments/files/17087823/SvtAv1EncApp.zip)
